### PR TITLE
V1-197 Similar courses functionality

### PR DESCRIPTION
**Pull request by zhuk-vezunchikz**
- Created migration for adding similar courses to the course.
- Integrated changes on the frontend side.

### CODE REVIEW

File: frontend/src/pages/DetailedCourse/DetailedCourseContainer.tsx [hunk 1]
```diff
@@ -4,14 +4,17 @@ import { useNavigate, useParams } from 'react-router';
 import { useApplyCourse, useGetCourseInfo } from 'api/courses';
 import { useGetClientCourseInfo } from 'api/myCourses';
 import { useGetProfile } from 'api/profile';
+import useGetMyCourses from 'api/myCourses/getClientCourses';
```
Comment (irina-sinitsyna) [inline frontend/src/pages/DetailedCourse/DetailedCourseContainer.tsx:7]:
```suggestion
import { useGetMyCourses } from 'api/myCourses';
```

File: frontend/src/pages/DetailedCourse/SimilarCourses/SimilarCourses.tsx [hunk 2]
```diff
@@ -17,31 +17,38 @@ import {
   SimilarCoursesTitle,
   SimilarCoursesWrapper,
 } from './styled';
+import transformRoute from '../../../utils/helpers/paths/transformRoute';
+import { PATHS } from '../../../constants/routes';
```
Comment (irina-sinitsyna) [inline frontend/src/pages/DetailedCourse/SimilarCourses/SimilarCourses.tsx:21]: please replace with absolute paths

File: frontend/src/pages/DetailedCourse/SimilarCourses/SimilarCourses.tsx [hunk 3]
```diff
@@ -17,31 +17,38 @@ import {
...
 
-const SimilarCourses: React.FC<ISimilarCourses> = ({ commonCourseData, windowWidth }) => (
+const SimilarCourses: React.FC<ISimilarCourses> = ({ similarCourses, windowWidth }) => (
```
Comment (irina-sinitsyna) [inline frontend/src/pages/DetailedCourse/SimilarCourses/SimilarCourses.tsx:23]:
```suggestion
const SimilarCourses: FC<ISimilarCourses> = ({ similarCourses, windowWidth }) => (
```

**Review by irina-sinitsyna — COMMENTED** [review]
image

**Review by ostapdans99 — COMMENTED** [review]
image

maybe we should display lessons and duration?

File: frontend/src/utils/helpers/similarCoursePreparing.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        acc.push(course as never);
+      }
+      return acc;
+    }, []);
+  }
+
+  return preparedSimilarCourses?.slice(0, countCoursesToDisplay);
+};
+
+export default similarCoursePreparing;
```
Comment (ostapdans99) [inline frontend/src/utils/helpers/similarCoursePreparing.ts:35]: is it good idea to do this checks in frontend?

File: frontend/src/api/myCourses/getClientCourses.ts [hunk 5]
```diff
@@ -26,4 +26,4 @@ const useGetMyCourses = (): UseQueryResult<Array<IClientCourse>, AxiosError> =>
   );
 };
 
-export default useGetMyCourses;
+export { useGetMyCourses };
```
Comment (irina-sinitsyna) [inline frontend/src/api/myCourses/getClientCourses.ts:29]: why not `export default useGetMyCourses;` ?

File: frontend/src/pages/DetailedCourse/DetailedCourseContainer.tsx [hunk 6]
```diff
@@ -4,7 +4,7 @@ import { useNavigate, useParams } from 'react-router';
 import { useApplyCourse, useGetCourseInfo } from 'api/courses';
 import { useGetClientCourseInfo } from 'api/myCourses';
 import { useGetProfile } from 'api/profile';
-import useGetMyCourses from 'api/myCourses/getClientCourses';
+import { useGetMyCourses } from 'api/myCourses/getClientCourses';
```
Comment (irina-sinitsyna) [inline frontend/src/pages/DetailedCourse/DetailedCourseContainer.tsx:7]: it's already exported from api/myCourses/index.ts, so you can import it like this: `import { useGetMyCourses } from 'api/myCourses';`

File: frontend/src/components/Course/CourseInfo/CourseInfo.tsx [hunk 7]
```diff
@@ -20,18 +20,23 @@ interface ICourseInfoProps {
...
-    </InfoItem>
+    {duration && (
+      <InfoItem>
+        <DurationIcon alt="lessons" src={clock} />
+        <InfoItemTextBox>
+          <InfoItemText>{duration}</InfoItemText>
+        </InfoItemTextBox>
+      </InfoItem>
+    )}
+
```
Comment (irina-sinitsyna) [inline frontend/src/components/Course/CourseInfo/CourseInfo.tsx:31]: remove empty line please

File: frontend/src/pages/DetailedCourse/SimilarCourses/SimilarCourses.tsx [hunk 8]
```diff
@@ -18,30 +21,42 @@ import {
   SimilarCoursesWrapper,
 } from './styled';
 
-const SimilarCourses: React.FC<ISimilarCourses> = ({ commonCourseData, windowWidth }) => (
+const DEFAULT_DISPLAYING_COURSES = 2;
```
Comment (irina-sinitsyna) [inline frontend/src/pages/DetailedCourse/SimilarCourses/SimilarCourses.tsx:24]: can we move it to constants/detailedCourse ?

File: frontend/src/pages/DetailedCourse/SimilarCourses/SimilarCourses.tsx [hunk 9]
```diff
@@ -18,30 +21,42 @@ import {
...
-              <CustomButton color="primary" variant="mediumOutlined">
-                {ButtonLabels.details}
-              </CustomButton>
-            </CourseActionsWrapper>
-          </CourseActionsBox>
-        </Course>
-      </SimilarCoursesItemWrapper>
+      {similarCourses.map(
+        (course, index) =>
+          index < DEFAULT_DISPLAYING_COURSES && (
```
Comment (irina-sinitsyna) [inline frontend/src/pages/DetailedCourse/SimilarCourses/SimilarCourses.tsx:32]: I suppose, this condition should be also replaced with a boolean constant

**Review by irina-sinitsyna — DISMISSED** [review]

File: backend/migrations-backdev/20220125143932-backdev_col.js [hunk 10]
```diff
@@ -751,6 +751,12 @@ const MATERIALS = [
         content: [
           'https://www.youtube.com/watch?v=xu87YWbr4X0&list=PLu8EoSxDXHP6CGK4YVJhL_VWetA865GOH&index=2&ab_channel=WesBos',
         ],
+        exercise: {
```
Comment (SaXaPhonist) [inline backend/migrations-backdev/20220125143932-backdev_col.js:754]: stale migration

**Pull request merged**